### ISSUE TITLE
Sonarr V4 Custom Script Error $sonarr_episodefile_path

### ISSUE THREAD

**Issue by Neptunespace** [Sonarr/Sonarr#5617]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Current Behavior

Sorry for my terrible English :(
My script always worked but not anymore.
For each new file added or modified, the script would write the path to the episode to a text file.

The script always tells me that it works when I do a test from the interface and in the sonarr log I see my message to indicate the test but the function of the script does not work anymore.
![image](https://user-images.githubusercontent.com/93327314/235896552-2ec88cbc-5c79-447c-ae64-41e949f6f33b.png)

My script :
```
#!/bin/bash
if [ ${sonarr_eventtype} = "Test" ]; then
  >&2 echo "[Debug] Testing script";
  exit;
fi
echo "$sonarr_episodefile_path" >> /config/Script/NewEpisode
```

### Expected Behavior

When a new episode is added, it writes the whole path in the text file: /config/Script/NewEpisode

### Steps To Reproduce

Nothing special, just add my script, good permission and try :(

### Environment

```markdown
- OS: Linux X86
- Sonarr: V4
- Docker Install: Yes, Linuxserver latest develop image available
- Using Reverse Proxy: Yes
- Browser: Chrome
```

### What branch are you running?

Develop

### Trace Logs?

```
[Error] ProcessProvider: Unable to set environment variable 'sonarr_release_indexer', value is null 

[v4.0.0.471] System.NullReferenceException: Object reference not set to an instance of an object.
   at NzbDrone.Common.Processes.ProcessProvider.Start(String path, String args, StringDictionary environmentVariables, Action`1 onOutputDataReceived, Action`1 onErrorDataReceived)

[Warn] NotificationService: Unable to send OnDownload notification to: MyScript

[v4.0.0.471] System.NullReferenceException: Object reference not set to an instance of an object.
   at NzbDrone.Common.Processes.ProcessProvider.Start(String path, String args, StringDictionary environmentVariables, Action`1 onOutputDataReceived, Action`1 onErrorDataReceived)
   at NzbDrone.Common.Processes.ProcessProvider.StartAndCapture(String path, String args, StringDictionary environmentVariables) in C:\BuildAgent\work\13f3e374fa512e16\src\NzbDrone.Common\Processes\ProcessProvider.cs:line 216
   at NzbDrone.Core.Notifications.CustomScript.CustomScript.ExecuteScript(StringDictionary environmentVariables) in C:\BuildAgent\work\13f3e374fa512e16\src\NzbDrone.Core\Notifications\CustomScript\CustomScript.cs:line 338
   at NzbDrone.Core.Notifications.CustomScript.CustomScript.OnDownload(DownloadMessage message) in C:\BuildAgent\work\13f3e374fa512e16\src\NzbDrone.Core\Notifications\CustomScript\CustomScript.cs:line 152
   at NzbDrone.Core.Notifications.NotificationService.Handle(EpisodeImportedEvent message) in C:\BuildAgent\work\13f3e374fa512e16\src\NzbDrone.Core\Notifications\NotificationService.cs:line 173
```

**Comment by bakerboy448** [comment]
Please provide the trace logs the bug report template requires and asks for not what error to be errors from the console.  If you need help figuring out how to get trace logs, add the wiki.

"V4" is not a valid version and your sonarr version can be found in system => status

**Comment by Neptunespace** [comment]
> Please provide the trace logs the bug report template requires and asks for not what error to be errors from the console. If you need help figuring out how to get trace logs, add the wiki.
> 
> "V4" is not a valid version and your sonarr version can be found in system => status

It's what you want ?
[sonarr.trace.txt](https://github.com/Sonarr/Sonarr/files/11382768/sonarr.trace.txt)

**Comment by CLHatch** [comment]
I noticed this in that  initial log...

`[v4.0.0.471] System.NullReferenceException: Object reference not set to an instance of an object.`

I'm on the same version, and I actually saw that `Object reference not set to an instance of an object.` message just the other night in the activity queue when hovering over the yellow arrow as a reason why the import was being rejected.  So possibly this is a wider issue, related to  #5608 ?

**Comment by augustuen** [comment]
> > Please provide the trace logs the bug report template requires and asks for not what error to be errors from the console. If you need help figuring out how to get trace logs, add the wiki.
> > "V4" is not a valid version and your sonarr version can be found in system => status
> 
> It's what you want ? [sonarr.trace.txt](https://github.com/Sonarr/Sonarr/files/11382768/sonarr.trace.txt)

Still doesn't show us what triggered the import. If you grabbed the torrent via a blackhole folder or by adding it to your torrent client it might be related to #5608. Or it might not, since we still don't know which version you're running.

**Comment by Neptunespace** [comment]
> > > Please provide the trace logs the bug report template requires and asks for not what error to be errors from the console. If you need help figuring out how to get trace logs, add the wiki.
> > > "V4" is not a valid version and your sonarr version can be found in system => status
> > 
> > 
> > It's what you want ? [sonarr.trace.txt](https://github.com/Sonarr/Sonarr/files/11382768/sonarr.trace.txt)
> 
> Still doesn't show us what triggered the import. If you grabbed the torrent via a blackhole folder or by adding it to your torrent client it might be related to #5608. Or it might not, since we still don't know which version you're running.

Ok you found the problem ;)
Blackhole > no output in my text file
Automatic Download > Working

I think this issue can be close and #5608 is the problem

**Comment by bakerboy448** [comment]
@mynameisbogdan same root cause?

**Comment by Neptunespace** [comment]
> Related lines of code:
> 
> https://github.com/Sonarr/Sonarr/blob/ca8b26138e3ebd793cc1a5fd51522ce3eda8a2e1/src/NzbDrone.Core/Notifications/CustomScript/CustomScript.cs#L141-L143
> 
> https://github.com/Sonarr/Sonarr/blob/ca8b26138e3ebd793cc1a5fd51522ce3eda8a2e1/src/NzbDrone.Common/Processes/ProcessProvider.cs#L130
> 
> And it's already fixed in #5559 by Q. :)

Oh :D

It's gonna be merge when you think ? :D
I need to reinstall sonarr with postgreSQL or something to patch this in futur ?

**Comment by Neptunespace** [comment]
@mynameisbogdan Thx men for your reply i wait this patch for next/futur release then :D
